src/Avalonia.DesignerSupport/Remote/HtmlTransport/SimpleWebSocketHttpServer.cs

```diff
@@ -325,16 +325,16 @@ async Task SendFrame(FrameType type, byte[] data, int offset, int length)
 
                 const byte endOfMessageBit = (byte)1u << 7;
                 header.Mask = (byte) (endOfMessageBit | ((byte) type & 0xf));
-                unsafe
```

Comment:
Should probably use span casts instead.

Reply:
It was actually what I tried first, but I found the resulting code not buying much:

```csharp
MemoryMarshal.AsBytes(new Span<WebSocketHeader>(&source, 1))
    .Slice(0, headerLength)
    .CopyTo(destination);
```
Not even saving the pointer usage to be compatible with .NET Standard.
That's basically the implementation of `Marshal.Copy`, so I didn't bother.

If the struct didn't have a variable size, we could simply use `MemoryMarshal.Write(destination, ref header)` instead: clear and concise. At that point, since we need 3 different structs, we're probably better writing directly to the destination buffer using `BinaryPrimitives.WriteX()`.